sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -185,8 +183,7 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         SetCatalogAndNamespace(catalogManager, None, Some(nameParts))
       } else {
         val CurrentCatalogAndNamespace(catalog, namespace) = nameParts
-        val ns = if (namespace.isEmpty) { None } else { Some(namespace) }
-        SetCatalogAndNamespace(catalogManager, Some(catalog.name()), ns)
+        SetCatalogAndNamespace(catalogManager, Some(catalog.name()), namespace)
```

Comment:
This will always set the catalog, even if it is already the current? Why not use a rule that doesn't fill in the catalog and returns None instead?

Reply:
The check is done here: https://github.com/apache/spark/blob/8556710409d9f2fbaee9dbf76a2ea70218316693/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala#L128-L137